Swap usage of `FastTree` binary to parallel `FastTreeMP`

jakereps opened the pull request:
Resolves #35 

~~~For some reason the `env` parameter on `subprocess.run` would not adjust the number of threads used by OpenMP. I had to resort to a context manager with direct os environment manipulation and that successfully limited the threads being used.~~~

jairideout commented:
Thanks for this feature @jakereps! We're close to wrapping up the release cycle for 2017.8 (scheduled for Wednesday) so we'll review/merge this one in the next cycle (2017.9). :tada:

jairideout commented on q2_phylogeny/_fasttree.py:
```diff
@@ -21,14 +22,22 @@ def run_command(cmd, output_fp, verbose=True):
         print(" ".join(cmd), end='\n\n')
+
+    if env is None:
+        env = os.environ.copy()
```
This env copy isn't necessary, you can just pass `env` through to `subprocess.run` if it's `None`. From `Popen` docs:

> If env is not None, it must be a mapping that defines the environment variables for the new process; these are used instead of the default behavior of inheriting the current process’ environment.

jairideout commented on q2_phylogeny/plugin_setup.py:
```diff
@@ -41,13 +42,15 @@
     },
-    parameter_descriptions={},
+    parameter_descriptions={
+        'n_threads': 'The number of threads.'
```
Can you add a note about `-1`? I tested this locally and that value seems to work as expected.

Pulling the text from mafft's `n_threads`:

```
(Use -1 to automatically use all available cores)
```

jairideout replied:
Can you also note the non-deterministic behavior when `FastTreeMP` is used, i.e. when `n_threads` is not equal to `1`? @gregcaporaso [requested that](https://github.com/qiime2/q2-phylogeny/issues/35#issuecomment-316770256) in the corresponding issue.

jairideout commented on q2_phylogeny/plugin_setup.py:
```diff
@@ -41,13 +42,15 @@
     function=q2_phylogeny.fasttree,
     inputs={'alignment': FeatureData[AlignedSequence]},
-    parameters={},
+    parameters={'n_threads': Int},
```
I realize we didn't do this with mafft's `n_threads` parameter (it was an oversight on my part), but I think the following type makes better use of the framework's type system:

```
Int % Range(-1, None)
```

jairideout commented on q2_phylogeny/_fasttree.py:
```diff
@@ -21,14 +22,22 @@ def run_command(cmd, output_fp, verbose=True):
+    env = os.environ.copy()
+    env.update({'OMP_NUM_THREADS': str(n_threads)})
+
+    cmd = ['FastTreeMP', '-quote', '-nt', aligned_fp]
```
Can you update the code so that `FastTree` is used when `n_threads=1`, and `FastTreeMP` is used when `n_threads` is any other value (see @gregcaporaso's [comment](https://github.com/qiime2/q2-phylogeny/issues/35#issuecomment-328945353))?

jakereps commented:
Thanks @jairideout! It'd been so long since the initial discussion that I completely forgot to cross-check what was said while implementing it. I've addressed all your comments!

jakereps commented on q2_phylogeny/plugin_setup.py:
```diff
@@ -50,8 +50,10 @@
+                     'runs the non-deterministic variant of `FastTree` '
+                     '(`FastTreeMP`), and may result in a different tree than '
+                     'single-threading. See '
+                     'http://www.microbesonline.org/fasttree/ for details. '
```
Do we want to include the anchor tag? `http://www.microbesonline.org/fasttree/#OpenMP`

jairideout replied:
Awesome, I couldn't find an anchor! Updated the PR with another squash commit.
